File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    all_lst = list('abcdefghijklmnopqrstuvwxyz')
    
    for k in s:
        current_index = all_lst.index(k)  
        steps = index 
        
        while steps > 0:
            current_index = (current_index + 1) % len(all_lst) 
            if all_lst[current_index] not in skip:
                steps -= 1  
        
        answer += all_lst[current_index]
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    public String solution(String s, String skip, int index) {
        StringBuilder answer = new StringBuilder();
        List<Character> allList = new ArrayList<>();
        
        // 알파벳 a-z를 리스트에 추가
        for (char c = 'a'; c <= 'z'; c++) {
            allList.add(c);
        }

        for (char k : s.toCharArray()) {
            int currentIndex = allList.indexOf(k);
            int steps = index;

            // 다음 인덱스를 찾기 위한 반복
            while (steps > 0) {
                currentIndex = (currentIndex + 1) % allList.size();
                // skip에 포함되지 않은 경우에만 steps 감소
                if (skip.indexOf(allList.get(currentIndex)) == -1) {
                    steps--;
                }
            }

            answer.append(allList.get(currentIndex));
        }
        
        return answer.toString();
    }
}
